pypeit/coadd2d.py

```diff
@@ -480,10 +489,12 @@ def coadd(self, only_slits=None, interp_dspat=True):
             maskdef_dict = self.get_maskdef_dict(slit_idx, ref_trace_stack)
 
             # weights
-            if not isinstance(self.use_weights, str) and self.use_weights.ndim > 2:
-                weights = self.use_weights[slit_idx]
-            else:
-                weights = self.use_weights
+            ## JFH I do not understand why this is here and am removing it.
+            #if not isinstance(self.use_weights, str):
+            #    weights = self.use_weights[slit_idx]
+            #else:
+            #    weights = self.use_weights
```

Comment:
Also the weights passed as a list in the coadd2d file do not work here.

Reply:
@debora-pe  I'm passing tests. So a test needs to be added to test this functionality since there is otherwise no way to see that it does not work. I still don't see the point of slit_idx if use_weights is a list.

Reply:
You are passing the tests because for echelle you are not setting `self.use_weights` when `weights = 'auto'`, it is therefore None and the code does not crash but does  not do what you want.
Also, look at how the weights are defined for echelle when `weights = 'auto'`. You can see that the weights are defined per order, i.e., each order has a different set of weights in each exposure. This is why `slit_idx` is needed here, but only for echelle.